### ISSUE TITLE
"ViewError: rule mark has no data. This is bug." error since v0.42.0

### ISSUE THREAD

**Issue by razsultana** [genome-spy/genome-spy#247]
First of all, kudos for creating such an awesome framework for genomic visualisations!
I have seen all the genome browsers since 2000 appear and tried all of them (I even wrote my own in 2000 - a poor man version of the UCSC browser) and this is the first time that I am really excited about a really nice, versatile, fast, beautiful and professionally written framework for genomic visualisations: WebGL, semantic zoom, lazy loading, embedding - wow!

I think it will become a de-facto standard for genomic visualisations as ggplot2 is for generic data visualisations.. 
Any other tool will have to at least match its capabilities. (I've seen http://gosling-lang.org, but I'm not so crazy about it).

I see that the framework is in continuous development and while trying to create my own visualisations (which happen to need some lollipop plots, I stumbled into this error (while using the _`spec=examples/lollipop.json`_) from the core package examples :

```
index.js:695 Uncaught ViewError: rule mark has no data. This is bug.
    at CL.createRenderCallback (index.js:695:4202)
    at CL.render (index.js:1106:2324)
    at FE.renderMark (index.js:1640:3293)
    at Qz.renderMark (index.js:1640:4444)
    at mt.render (index.js:1633:18664)
    at ys.render (index.js:1667:14269)
    at YE.computeLayout (index.js:1672:7644)
    at ResizeObserver.t (index.js:1672:507
```

I had very little exposure to javascript programming and development (although I am getting into it now), so I am not well equiped to debug this error. I think I first found it in version  0.45 (which was current at the time) and when I switched to version 0.35 - which is the one shown in the website-examples, it went away.

It is still present in the last version - 0.50.0 and I have gone back and tried each version until I got to the one that doesn't generate this error: 0.41.0, so it seems that it was first introduced in 0.42.0 (maybe some winter holiday coding ?)

It also appears in other examples from the website-examples package, for example _`spec=ASCAT/ascat.json`_, but only starting in version 0.45.0 and co-occurring with another error that popped up in the **_ASCAT_** example (which I will document in a different issue) - that error started to pop at 0.44.0, but is only associated with this one from 0.45.0 :

```
index.js:1831 TypeError: Cannot read properties of undefined (reading 'length')
    at jn (index.js:1:2985)
    at Mz (index.js:1740:1975)
    at Bz (index.js:1740:1354)
    at YE (index.js:1740:665)
    at Iz (index.js:1740:822)
    at get scale (index.js:1740:11095)
    at tM (index.js:222:54347)
    at GN.initializeEncoders (index.js:817:2476)
    at index.js:1831:2374
    at Array.forEach (<anonymous>)
index.js:820 Uncaught ViewError: rect mark has no data. This is bug.
    at BN.createRenderCallback (index.js:820:3853)
    at BN.render (index.js:820:8963)
    at C4.renderMark (index.js:1760:3313)
    at b$.renderMark (index.js:1760:4464)
    at bt.render (index.js:1740:19373)
    at io.render (index.js:1787:605)
    at Es.render (index.js:1787:14226)
    at H4.computeLayout (index.js:1831:6327)
    at ResizeObserver.n (index.js:1831:7750)
```

The first error seems to got fixed by version 0.47.0, but the second one still appears in version 0.50.0 :

```
index.js:850 Uncaught ViewError: rect mark has no data. This is bug.
    at Hz.createRenderCallback (index.js:850:4114)
    at Hz.render (index.js:850:9279)
    at dS.renderMark (index.js:1820:3375)
    at CH.renderMark (index.js:1820:4526)
    at bt.render (index.js:1800:28700)
    at uo.render (index.js:1847:1378)
    at Ls.render (index.js:1847:16307)
    at BS.computeLayout (index.js:1907:6562)
    at ResizeObserver.n (index.js:1907:8021)
```

The visualisations actually work (I can't tell if anything is out of order), but the error message is preventing deploying this visualisation into any "production" context.
Hope this helps!

**Comment by tuner** [comment]
Thanks, @razsultana

I'm happy that you found the toolkit useful. If you encounter any other issues (bugs, improvement ideas, etc.), just let me know!

That error appeared after I did some refactoring quite some time ago, and it is related to the initialization phase of the visualization. In practice, it doesn't seem to cause any incorrect behavior. Still, I wanted to review all the logic rather than just silence the message. I've been busy with other issues, but I could check that in the coming days.

**Comment by tuner** [comment]
Should now be fixed in [v0.50.1](https://github.com/genome-spy/genome-spy/releases/tag/v0.50.1).
